### ISSUE TITLE
date_range default calendar day return

### ISSUE THREAD

**Issue by zxymark221** [pydata/pandas#12705]
#### Code Executed:
```
# 0.000016 second less than 24 hours, acrossing 2 calendar day.
print pandas.date_range(start='2016-03-22 12:19:49.053501', end='2016-03-23 12:19:49.053485')
```
#### Output of different versions.
- __Prior to 0.17__<br>
Prior to 0.17 versions, I tested 0.16 and 0.13, they both return two days.<br>
`DatetimeIndex(['2016-03-22 12:19:49.053501', '2016-03-23 12:19:49.053501'], dtype='datetime64[ns]', freq='D', tz=None)`
- __V0.17 & 0.18__<br>
Returns one day only<br>
`DatetimeIndex(['2016-03-22 12:19:49.053501'], dtype='datetime64[ns]', freq='D')`

On the [documentation](http://pandas.pydata.org/pandas-docs/stable/generated/pandas.date_range.html?highlight=date_range#pandas.date_range), the __freq__ is set to be __calendar daily__ by default, the test above covers two calendar days, does it make more sense to return two days like the earlier versions do?

**Comment by zxymark221** [comment]
@jreback Thanks for your reply.

What I am trying to say is: I am expecting the code execute above should return two days(both 2016-03-22 and 2016-03-23) even though the start and end time I specified above is less than 24 hrs. This is because the freq setting in date_range is by default *calendar daily*. From my perspective, calendar daily is different from 24 hours, otherwise, why should we call it __calendar__ daily.

**Comment by jreback** [comment]
@zxymark221 I don't think so. you asked for 1 Day from the start effectively as stop time < start time + 1 day:

```
In [3]: start = Timestamp('2016-03-22 12:19:49.053501')

In [4]: stop = Timestamp('2016-03-23 12:19:49.053485')

In [6]: start+pd.offsets.Day()
Out[6]: Timestamp('2016-03-23 12:19:49.053501')

In [8]: stop-start+pd.offsets.Day()
Out[8]: Timedelta('1 days 23:59:59.999984')

In [9]: start+pd.offsets.Day() > stop
Out[9]: True
```

**Comment by zxymark221** [comment]
@jreback Thank you. Understood. By default, freq is 24 hours. 

I've been using 0.16 version for a long time, and the idea of calendar daily [here](http://pandas.pydata.org/pandas-docs/stable/generated/pandas.date_range.html?highlight=date_range#pandas.date_range) made me think it will return all the dates the range has been through.
